ng new command fails when schematics are installed in a non-flattened structure

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION YOUR ISSUE MIGHT BE CLOSED WITHOUT INVESTIGATING
-->
### Bug Report or Feature Request (mark with an `x`)
```
- [ x] bug report -> please search issues before submitting
- [ ] feature request
```

### Area
```
- [ ] devkit
- [x ] schematics
```

### Versions
<!--
Output from: `node --version` and `npm --version`.
  Windows (7/8/10). Linux (incl. distribution). macOS (El Capitan? Sierra? High Sierra?)
-->
node 8.9.3
npm 5.5.1
pnpm 1.25.1
macOS High Sierra

### Repro steps
<!--
Simple steps to reproduce this bug.
Please include: commands run, packages added, related code changes.
A link to a sample repo would help too.
-->
```
npm i -g pnpm@1.25.1
pnpm i -g @angular/cli@1.6.3
ng new test-angular-cli
```
### The log given by the failure
<!-- Normally this include a stack trace and some more information. -->
```
$ ng new test-angular-cli
Cannot find module 'typescript'
Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/aecz/.nvm/versions/node/v8.9.3/pnpm-global/1/node_modules/.registry.npmjs.org/@schematics/angular/0.1.11/93b61b37ad36596a0cae6bf7dde8b984/node_modules/@schematics/angular/application/index.js:11:12)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
```
We can see that the error is in the package `@schematics/angular`. A file is importing `typescript` but `package.json` is not declaring it.

### Desired functionality
<!--
What would like to see implemented?
What is the usecase?
-->
Direct dependencies used by `@schematics/angular` should be included in the corresponding `package.json` instead of relying on the flattening of dependencies done by `NPM`. This could change and also the package providing `typescript` indirectly could stop using it and `@schematics/angular` would throw an error.
Fixing this Angular CLI would also work with `pnpm` like it does with `npm` and `yarn`.

### Mention any other details that might be useful
<!-- Please include a link to the repo if this is related to an OSS project. -->
A tool like `depcheck` could be used to detect those errors like it is done in Angular CLI project.

--- a/packages/angular_devkit/build_optimizer/package.json
+++ b/packages/angular_devkit/build_optimizer/package.json
@@ -14,7 +14,7 @@
   "dependencies": {
     "loader-utils": "^1.1.0",
     "source-map": "^0.5.6",
-    "typescript": "~2.6.1",
+    "typescript": "~2.6.2",
     "webpack-sources": "^1.0.1"
   }
 }

--- a/packages/angular_devkit/core/package.json
+++ b/packages/angular_devkit/core/package.json
@@ -13,6 +13,7 @@
   "dependencies": {
     "ajv": "~5.5.1",
     "chokidar": "^1.7.0",
-    "source-map": "^0.5.6"
+    "source-map": "^0.5.6",
+    "rxjs": "^5.5.6"
   }
 }

--- a/packages/angular_devkit/schematics/package.json
+++ b/packages/angular_devkit/schematics/package.json
@@ -16,10 +16,10 @@
     "schematics"
   ],
   "dependencies": {
-    "@angular-devkit/core": "0.0.0",
     "@ngtools/json-schema": "^1.1.0",
-    "@schematics/schematics": "0.0.0",
-    "minimist": "^1.2.0",
     "rxjs": "^5.5.6"
+  },
+  "peerDependencies": {
+    "@angular-devkit/core": "0.0.0"
   }
 }

--- a/packages/angular_devkit/schematics_cli/package.json
+++ b/packages/angular_devkit/schematics_cli/package.json
@@ -21,6 +21,6 @@
     "@angular-devkit/schematics": "0.0.0",
     "@schematics/schematics": "0.0.0",
     "minimist": "^1.2.0",
-    "rxjs": "^5.5.2"
+    "rxjs": "^5.5.6"
   }
 }

--- a/packages/schematics/angular/package.json
+++ b/packages/schematics/angular/package.json
@@ -12,9 +12,10 @@
   },
   "schematics": "./collection.json",
   "dependencies": {
-    "@angular-devkit/core": "0.0.0"
+    "typescript": "~2.6.2"
   },
   "peerDependencies": {
+    "@angular-devkit/core": "0.0.0",
     "@angular-devkit/schematics": "0.0.0"
   }
 }

--- a/packages/schematics/package-update/package.json
+++ b/packages/schematics/package-update/package.json
@@ -12,11 +12,13 @@
     "preinstall": "echo DO NOT INSTALL THIS PROJECT, ONLY THE ROOT PROJECT. && exit 1"
   },
   "schematics": "./collection.json",
+  "dependencies": {
+    "semver": "^5.3.0",
+    "semver-intersect": "^1.1.2",
+    "rxjs": "^5.5.6"
+  },
   "peerDependencies": {
     "@angular-devkit/core": "0.0.0",
-    "@angular-devkit/schematics": "0.0.0",
-    "rxjs": "^5.5.6",
-    "semver": "^5.3.0",
-    "semver-intersect": "^1.1.2"
+    "@angular-devkit/schematics": "0.0.0"
   }
 }

--- a/packages/schematics/schematics/package.json
+++ b/packages/schematics/schematics/package.json
@@ -14,7 +14,6 @@
   "schematics": "./collection.json",
   "peerDependencies": {
     "@angular-devkit/core": "0.0.0",
-    "@angular-devkit/schematics": "0.0.0",
-    "rxjs": "^5.5.6"
+    "@angular-devkit/schematics": "0.0.0"
   }
 }

